airbyte-integrations/connectors/source-linnworks/source_linnworks/source.py

```diff
@@ -0,0 +1,324 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+
+from abc import ABC, abstractmethod
+from datetime import date
+from typing import (Any, Iterable, List, Mapping, MutableMapping, Optional,
+                    Tuple, Union)
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum
+import requests
+from airbyte_cdk.models.airbyte_protocol import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth.core import HttpAuthenticator
+from airbyte_cdk.sources.streams.http.requests_native_auth import \
+    Oauth2Authenticator
+from requests.auth import AuthBase
+
+from .util import normalize
+import json
+
+
+class LinnworksStream(HttpStream, ABC):
+    http_method = "POST"
+
+    @property
+    def url_base(self) -> str:
+        return self.authenticator.get_server()
+
+    def __init__(self, authenticator: Union[AuthBase, HttpAuthenticator] = None, start_date: str = None):
+        super().__init__(authenticator=authenticator)
+
+        self._authenticator = authenticator
+        self.start_date = start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        if response.status_code != requests.codes.ok:
+            return None
+
+        json = response.json()
+        if not isinstance(json, list):
+            json = [json]
+        for record in json:
+            yield normalize(record)
```

Comment:
Linnworks API uses `TitleCase` and `camelCase` interchangeably. They both ensure name uniqueness only in a case-sensitive environment. In a case-insensitive environment, `camelCase` and `TitleCase` may lead to name collision issues.

I'm using Postgres destination, which is case-insensitive and `snake_case` is the preferred casing. What's your suggestion?

Reply:
I think this case matter can be handled at normalization step. Basic or custom normalization in Airbyte will allow you to solve this for postgres and obtain the schema you want in your destination. In my opinion you should not alter the original Linnworks' schema: it might be misleading for users using your connectors that might expect output data similar to what they would get calling the API. Moreover it might be a maintenance challenge in the future. We expect the output data of our connectors to be as raw as possible, here you are making a small transformation that should occur later (in the transformation) and outside of the scope of the connector.